rms_fe/src/views/ArticleDetailsPage/ArticleRecommendations.tsx

```diff
@@ -108,25 +110,101 @@ const AuthorPopover: React.FC<{ author: Author }> = ({ author }) => {
     );
 };
 
-const AuthorLinks: React.FC<{ author: Author }> = ({ author }) => (
-    <>
-        <AuthorLink icon={<LinkOutlined />} url={author.scopus_url} label="Scopus Profile" />
-        <AuthorLink icon={<LinkOutlined />} url={"#"} label="DBLP Profile" />
-        <AuthorLink icon={<LinkOutlined />} url={"#"} label="Scolar Profile" />
-    </>
-);
 
-const AuthorLink: React.FC<{ icon: React.ReactNode; url?: string; label: string }> = ({ icon, url, label }) => {
-    if (!url) return null;
+const useDblpAuthor = (author_name: string) => {
+    return useQuery({ queryKey: ["author_dblp", author_name], queryFn: () => getAuthorDBLP(author_name), staleTime: 5 * 60 * 1000 })
+}
+
+const useScholarAuthor = (author_name: string) => {
+    return useQuery({ queryKey: ["author_scholar", author_name], queryFn: () => getAuthorScholar(author_name), staleTime: 5 * 60 * 1000 })
+}
+
+const AuthorLinks: React.FC<{ author: Author }> = ({ author }) => {
+
+    const authorName = author.given_name + " " + author.surname;
+    const dblpAuthor = useDblpAuthor(authorName);
+    const scholarAuthor = useScholarAuthor(authorName);
+
```

Comment:
To nicely enable preloading of these information, I would move the queries to the parent component, and just pass the whole object that you get from the useQuery

Reply:
Preloading information for all results might result in throttling due to the DBLP and Scholar API limitations. Considering the potential high number of authors in recommended articles, and that probably we won't need all of them, preloading may not be the best idea.